### ISSUE TITLE
After OTA nightly update koreader (nightly nor stable) doesn't work:(

### ISSUE THREAD

**Issue by mwozny123** [koreader/koreader#1948]
* KOReader version:
stable and current nightly 

* Device:
Pocketbook Touch Lux 3

Log:

> ./luajit: error loading module 'libs/libkoreader-input' from file './libs/libkoreader-input.so':
	/usr/lib/libcurl.so.4: undefined symbol: SSLv2_client_method
stack traceback:
	[C]: at 0x00044bd9
	[C]: at 0x0003dc81
	[C]: in function 'require'
	frontend/device/input.lua:3: in main chunk
	[C]: in function 'require'
	frontend/device/pocketbook/device.lua:81: in function 'init'
	frontend/device.lua:39: in main chunk
	[C]: in function 'require'
	frontend/ui/uimanager.lua:1: in main chunk
	[C]: in function 'require'
	./reader.lua:96: in main chunk
	[C]: at 0x0000be01

[crash.log](https://github.com/koreader/koreader/files/208992/crash.log.txt)

#### Issue
After OTA nightly update I'm not able to run Koreder. 
I've tried to download, extract and run again but it doesn't work neither stable nor nightly build.

#### Steps to reproduce
I'd nightly 2015.11 version then today updated (via OTA) to recent nightly. 
Koreader was unable to start.
I'd removed koreader for device.
Download, extracted and run Stable bulid but it didn't work - removed it.
Download, extracted and run Latest Nightly build but it also didn't work :(

**Comment by NiLuJe** [comment]
Err, something weird is happening...

• OpenSSL 1.0.2g dropped ssl2 support.
• We're not using OpenSSL 1.0.2. And even if we did, we build everything against it, so that wouldn't happen.
• Why the hell is libkoreader-input pulling ing lib curl??!

**Comment by NiLuJe** [comment]
Hypothesis:

Given the same thing from the latest PB OTA:
```
readelf -d applications/koreader/libs/libkoreader-input.so | ag NEEDED
 0x00000001 (NEEDED)                     Shared library: [libinkview.so]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
```

I'd say libinkview is the one pulling curl, and, for some incomprehensible reason (as-needed is your friend), that is pulling in curl. curl, in turns, depends on OpenSSL (for good reason, this time), but given our RPATH, our own OpenSSL copy is picked first.

And, for some reason (did someone update OpenSSL recently?), our copy doesn't support ssl2, which upsets the curl copy from the PB system.

**Comment by NiLuJe** [comment]
```
readelf -a applications/koreader/libs/libssl.so.1.0.0 | ag SSLv2
    93: 0001c289    12 FUNC    GLOBAL DEFAULT   10 SSLv23_client_method
   105: 0001bb35    12 FUNC    GLOBAL DEFAULT   10 SSLv23_method
   118: 0001bb75    12 FUNC    GLOBAL DEFAULT   10 SSLv23_server_method
```

Which matches the output from my own, SSL2-less OpenSSL builds.

**Comment by NiLuJe** [comment]
TL;DR: Guess we need to configure OpenSSL w/ enable-ssl2 for the PB builds?

**Comment by NiLuJe** [comment]
What's strange is, we haven't touched OpenSSL in a while, so...

Did you update your PB firmware recently, instead?

**Comment by mwozny123** [comment]
Well some time ago there was a FW update, but after it I had no problem at all with koreader.

Until today I used the nightly 201511-something version.
But today I'd checked OTA for Koreader update and I've updated to latest nightly. 
New OTA Pocketbook version also started with 201511...

Since then (after Koreader restart) it doesn't start.

I tried (using PC) to remove it, and download directly from web page, extract and run, but it didn't help.
Neither for Stable nor Nightly. 

I've just check FW update and I'm up to date with Pocketbook FW.

**Comment by houqp** [comment]
It does really look like an issue caused by firmware upgrade... We did upgrade openssl to OpenSSL_1_0_1s after the stable build. But it looks like the stable is also broken for @mwozny123 . So yeah, I think we might have to enable sslv2 fo PB build. It's sad because it's a broken protocol :(

**Comment by mwozny123** [comment]
I did some more tests - checking which version works and which doesn't:
1) koreader-pocketbook-arm-obreey-linux-gnueabi-v2014.11-49-test 
starts, is running but display is deformed - can't read anything - but this is very old no worries

2) koreader-pocketbook-arm-obreey-linux-gnueabi-v2014.11-415-g5dc899f 
works, was able to open pdf, in fast test didn't noticed any problems

3) koreader-pocketbook-arm-obreey-linux-gnueabi-v2014.11-421-g6c51645
works, was able to open pdf, in fast test didn't noticed any problems

4) koreader-pocketbook-arm-obreey-linux-gnueabi-v2015.11-11-g16509e2
works, was able to open pdf, in fast test didn't notice any problems

5) koreader-pocketbook-arm-obreey-linux-gnueabi-v2015.11-118-ge71863a
works, was able to open pdf, in fast test didn't notice any problems, but in log there's something strange - stacktrace:
[crash.log](https://github.com/koreader/koreader/files/210612/crash.log.txt)

6) koreader-pocketbook-arm-obreey-linux-gnueabi-v2015.11-175-g7b11491
works, was able to open pdf, in fast test didn't notice any problems,

7) koreader-pocketbook-arm-obreey-linux-gnueabi-v2015.11-253-gcf9dc26
doesn't work "undefined symbol: SSLv2_client_method":
[crash.log](https://github.com/koreader/koreader/files/210599/crash.log.txt)

There must be something wrong with my yesterdays test with stable version - maybe corrupted file transfer? Please don't bother that.
So no 6) was the last one which worked.
Was there a ssl change between version from 6) and 7) ?

**Comment by NiLuJe** [comment]
Most likely, yes.
